# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424200246 18000
#      Tue Feb 17 14:10:46 2015 -0500
# Node ID bf25e9f7c22ad4695c7a363a869cf626db59d72c
# Parent  9861618b4c5fe96ed04d7a7a6c5b69004c6c8408
use FLTK_CFLAGS and FLTK_LIBS instead of GRAPHICS_CFLAGS and GRAPHICS_LIBS

* common.mk, configure.ac, toplev.cc, __osmesa_print__.cc,
libinterp/dldfcn/module-files, oct-conf.in.h: Use FLTK_CFLAGS and
FLTK_LIBS instead of GRAPHICS_CFLAGS and GRAPHICS_LIBS.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -220,33 +220,33 @@ FFTW3_LIBS = @FFTW3_LIBS@
 FFTW3F_CPPFLAGS = @FFTW3F_CPPFLAGS@
 FFTW3F_LDFLAGS = @FFTW3F_LDFLAGS@
 FFTW3F_LIBS = @FFTW3F_LIBS@
 
 FFTW_XCPPFLAGS = $(FFTW3_CPPFLAGS) $(FFTW3F_CPPFLAGS)
 FFTW_XLDFLAGS = $(FFTW3_LDFLAGS) $(FFTW3F_LDFLAGS)
 FFTW_XLIBS = $(FFTW3_LIBS) $(FFTW3F_LIBS)
 
+FLTK_CFLAGS = @FLTK_CFLAGS@
+FLTK_LIBS = @FLTK_LIBS@
+
 ## Alias CPPFLAGS to CFLAGS.
 ## This is closer to the true meaning of `pkg-config --cflags` output.
 FONTCONFIG_CPPFLAGS = @FONTCONFIG_CFLAGS@
 FONTCONFIG_CFLAGS = @FONTCONFIG_CFLAGS@
 
 ## Alias CPPFLAGS to CFLAGS.
 FT2_CPPFLAGS = @FT2_CFLAGS@
 FT2_CFLAGS = @FT2_CFLAGS@
 FT2_LIBS = @FT2_LIBS@
 
 GLPK_CPPFLAGS = @GLPK_CPPFLAGS@
 GLPK_LDFLAGS = @GLPK_LDFLAGS@
 GLPK_LIBS = @GLPK_LIBS@
 
-GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
-GRAPHICS_LIBS = @GRAPHICS_LIBS@
-
 GL2PS_LIBS = @GL2PS_LIBS@
 
 HDF5_CPPFLAGS = @HDF5_CPPFLAGS@
 HDF5_LDFLAGS = @HDF5_LDFLAGS@
 HDF5_LIBS = @HDF5_LIBS@
 
 JAVA_CPPFLAGS = @JAVA_CPPFLAGS@
 JAVA_LIBS = @JAVA_LIBS@
@@ -551,26 +551,26 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_FFLAGS%|\"${FFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_CPPFLAGS%|\"${FFTW3_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_LDFLAGS%|\"${FFTW3_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3_LIBS%|\"${FFTW3_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_CPPFLAGS%|\"${FFTW3F_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_LDFLAGS%|\"${FFTW3F_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FFTW3F_LIBS%|\"${FFTW3F_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FLIBS%|\"${FLIBS}\"|" \
+  -e "s|%OCTAVE_CONF_FLTK_CFLAGS%|\"${FLTK_CFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_FLTK_LIBS%|\"${FLTK_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_FONTCONFIG_CFLAGS%|\"${FONTCONFIG_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FPICFLAG%|\"${FPICFLAG}\"|" \
   -e "s|%OCTAVE_CONF_FT2_CFLAGS%|\"${FT2_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_FT2_LIBS%|\"${FT2_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_CPPFLAGS%|\"${GLPK_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_LDFLAGS%|\"${GLPK_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_GLPK_LIBS%|\"${GLPK_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_GNUPLOT%|\"${GNUPLOT}\"|" \
-  -e "s|%OCTAVE_CONF_GRAPHICS_CFLAGS%|\"${GRAPHICS_CFLAGS}\"|" \
-  -e "s|%OCTAVE_CONF_GRAPHICS_LIBS%|\"${GRAPHICS_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_HDF5_CPPFLAGS%|\"${HDF5_CPPFLAGS}\"|" | \
   $(SED) -e "s|%OCTAVE_CONF_HDF5_LDFLAGS%|\"${HDF5_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_HDF5_LIBS%|\"${HDF5_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_INCLUDEDIR%|\"${includedir}\"|" \
   -e "s|%OCTAVE_CONF_LAPACK_LIBS%|\"${LAPACK_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_LD_CXX%|\"${LD_CXX}\"|" \
   -e "s|%OCTAVE_CONF_LDFLAGS%|\"${LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_LD_STATIC_FLAG%|\"${LD_STATIC_FLAG}\"|" \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1167,18 +1167,18 @@ AC_ARG_WITH([opengl],
    fi],
   [check_opengl=yes])
 
 ## Check for OpenGL library
 if test $check_opengl = yes; then
   OCTAVE_CHECK_LIB_OPENGL
 fi
 
-GRAPHICS_LIBS=
-GRAPHICS_CFLAGS=
+FLTK_LIBS=
+FLTK_CFLAGS=
 
 if test -z "$OPENGL_LIBS"; then
   if test $check_opengl = yes; then
     native_graphics=no
     warn_fltk_opengl="OpenGL libs (GL and GLU) not found.  Native graphics will be disabled."
     OCTAVE_CONFIGURE_WARNING([warn_fltk_opengl])
   fi
 fi
@@ -1316,45 +1316,44 @@ if test -n "$OPENGL_LIBS"; then
         CFLAGS="$save_CFLAGS"
         ])
       if test $octave_cv_fltk_opengl_support = no; then
         warn_fltk_opengl="FLTK does not have OpenGL support.  Native graphics will be disabled."
       else
         AC_DEFINE(HAVE_FLTK, 1, [Define to 1 if FLTK is available.])
       fi
 
-      if test -z "$warn_fltk_opengl"; then
-        GRAPHICS_CFLAGS="$FLTK_CFLAGS"
-        GRAPHICS_LIBS="$FLTK_LDFLAGS"
-      else
+      if test -n "$warn_fltk_opengl"; then
         native_graphics=no
         OCTAVE_CONFIGURE_WARNING([warn_fltk_opengl])
       fi
     fi
   fi
 fi
 
-AC_SUBST(GRAPHICS_CFLAGS)
-AC_SUBST(GRAPHICS_LIBS)
+AC_SUBST(FLTK_CFLAGS)
+AC_SUBST(FLTK_LIBS)
 
 ## Check for gl2ps which is required for printing with OpenGL graphics
 if test $native_graphics = yes; then
   AC_CHECK_HEADERS([gl2ps.h],
     [GL2PS_LIBS="-lgl2ps"],
     [warn_gl2ps="gl2ps library not found.  OpenGL printing is disabled."
      OCTAVE_CONFIGURE_WARNING([warn_gl2ps])])
 fi
 
 AC_SUBST(GL2PS_LIBS)
 
-## Check for OSMesa which is needed for offscreen rendering with OpenGL
-## FIXME: Check for version >= 9.0.0
-OCTAVE_CHECK_LIB(OSMesa, OSMesa,
-  [OSMesa library not found.  Offscreen rendering with OpenGL will be disabled.],
-  [GL/osmesa.h], [OSMesaCreateContext])
+if test -n "$OPENGL_LIBS"; then
+  ## Check for OSMesa which is needed for offscreen rendering with OpenGL
+  ## FIXME: Check for version >= 9.0.0
+  OCTAVE_CHECK_LIB(OSMesa, OSMesa,
+    [OSMesa library not found.  Offscreen rendering with OpenGL will be disabled.],
+    [osmesa.h GL/osmesa.h], [OSMesaCreateContext])
+fi
 
 ### Start determination of shared vs. static libraries
 
 ## Use -static if compiling on Alpha OSF/1 1.3 systems.
 case $canonical_host_type in
   alpha*-dec-osf1.3)
     LD_STATIC_FLAG=-static
   ;;
@@ -3020,25 +3019,25 @@ Octave is now configured for $canonical_
   CXSPARSE libraries:          $CXSPARSE_LIBS
   DL libraries:                $DL_LIBS
   FFTW3 CPPFLAGS:              $FFTW3_CPPFLAGS
   FFTW3 LDFLAGS:               $FFTW3_LDFLAGS
   FFTW3 libraries:             $FFTW3_LIBS
   FFTW3F CPPFLAGS:             $FFTW3F_CPPFLAGS
   FFTW3F LDFLAGS:              $FFTW3F_LDFLAGS
   FFTW3F libraries:            $FFTW3F_LIBS
+  FLTK CFLAGS:                 $FLTK_CFLAGS
+  FLTK libraries:              $FLTK_LIBS
   fontconfig CFLAGS:           $FONTCONFIG_CFLAGS
   fontconfig libraries:        $FONTCONFIG_LIBS
   FreeType2 CFLAGS:            $FT2_CFLAGS
   FreeType2 libraries:         $FT2_LIBS
   GLPK CPPFLAGS:               $GLPK_CPPFLAGS
   GLPK LDFLAGS:                $GLPK_LDFLAGS
   GLPK libraries:              $GLPK_LIBS
-  graphics CFLAGS:             $GRAPHICS_CFLAGS
-  graphics libraries:          $GRAPHICS_LIBS
   HDF5 CPPFLAGS:               $HDF5_CPPFLAGS
   HDF5 LDFLAGS:                $HDF5_LDFLAGS
   HDF5 libraries:              $HDF5_LIBS
   Java home:                   $JAVA_HOME
   Java JVM path:               $JAVA_LDPATH
   Java CPPFLAGS:               $JAVA_CPPFLAGS
   Java libraries:              $JAVA_LIBS
   LAPACK libraries:            $LAPACK_LIBS
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1339,26 +1339,26 @@ specified option.\n\
     { false, "FFLAGS", OCTAVE_CONF_FFLAGS },
     { false, "FFTW3_CPPFLAGS", OCTAVE_CONF_FFTW3_CPPFLAGS },
     { false, "FFTW3_LDFLAGS", OCTAVE_CONF_FFTW3_LDFLAGS },
     { false, "FFTW3_LIBS", OCTAVE_CONF_FFTW3_LIBS },
     { false, "FFTW3F_CPPFLAGS", OCTAVE_CONF_FFTW3F_CPPFLAGS },
     { false, "FFTW3F_LDFLAGS", OCTAVE_CONF_FFTW3F_LDFLAGS },
     { false, "FFTW3F_LIBS", OCTAVE_CONF_FFTW3F_LIBS },
     { false, "FLIBS", OCTAVE_CONF_FLIBS },
+    { false, "FLTK_CFLAGS", OCTAVE_CONF_FLTK_CFLAGS },
+    { false, "FLTK_LIBS", OCTAVE_CONF_FLTK_LIBS },
     { false, "FONTCONFIG_CFLAGS", OCTAVE_CONF_FONTCONFIG_CFLAGS },
     { false, "FPICFLAG", OCTAVE_CONF_FPICFLAG },
     { false, "FT2_CFLAGS", OCTAVE_CONF_FT2_CFLAGS },
     { false, "FT2_LIBS", OCTAVE_CONF_FT2_LIBS },
     { false, "GLPK_CPPFLAGS", OCTAVE_CONF_GLPK_CPPFLAGS },
     { false, "GLPK_LDFLAGS", OCTAVE_CONF_GLPK_LDFLAGS },
     { false, "GLPK_LIBS", OCTAVE_CONF_GLPK_LIBS },
     { false, "GNUPLOT", OCTAVE_CONF_GNUPLOT },
-    { false, "GRAPHICS_CFLAGS", OCTAVE_CONF_GRAPHICS_CFLAGS },
-    { false, "GRAPHICS_LIBS", OCTAVE_CONF_GRAPHICS_LIBS },
     { false, "HDF5_CPPFLAGS", OCTAVE_CONF_HDF5_CPPFLAGS },
     { false, "HDF5_LDFLAGS", OCTAVE_CONF_HDF5_LDFLAGS },
     { false, "HDF5_LIBS", OCTAVE_CONF_HDF5_LIBS },
     { false, "LAPACK_LIBS", OCTAVE_CONF_LAPACK_LIBS },
     { false, "LDFLAGS", OCTAVE_CONF_LDFLAGS },
     { false, "LD_CXX", OCTAVE_CONF_LD_CXX },
     { false, "LD_STATIC_FLAG", OCTAVE_CONF_LD_STATIC_FLAG },
     { false, "LEX", OCTAVE_CONF_LEX },
diff --git a/libinterp/dldfcn/__osmesa_print__.cc b/libinterp/dldfcn/__osmesa_print__.cc
--- a/libinterp/dldfcn/__osmesa_print__.cc
+++ b/libinterp/dldfcn/__osmesa_print__.cc
@@ -31,19 +31,23 @@ from git://anongit.freedesktop.org/mesa/
 #include "unwind-prot.h"
 
 #include "defun-dld.h"
 #include "gl-render.h"
 #include "gl2ps-renderer.h"
 #include "graphics.h"
 #include "gripes.h"
 
-#ifdef HAVE_OSMESA
+#if defined (HAVE_OSMESA_H)
+#include "osmesa.h"
+#elif defined (HAVE_GL_OSMESA_H)
 #include "GL/osmesa.h"
+#endif
 
+#if defined (HAVE_OSMESA) && defined (HAVE_OPENGL)
 static void
 close_fcn (FILE *f)
 {
   gnulib::fclose (f);
 }
 #endif
 
 DEFUN_DLD(__osmesa_print__, args, ,
@@ -68,17 +72,17 @@ Don't render text.\n\
 @end table\n\
 \n\
 The second method doesn't use gl2ps and returns a RGB image in @var{img} instead.\n\
 \n\
 @end deftypefn")
 {
   octave_value_list retval;
 
-#ifndef HAVE_OSMESA
+#if ! defined (HAVE_OSMESA)
   gripe_disabled_feature ("__osmesa_print__", "Offscreen rendering");
 #else
 
   int nargin = args.length ();
 
   if (! (nargin == 1 || nargin == 3))
     {
       print_usage ();
diff --git a/libinterp/dldfcn/module-files b/libinterp/dldfcn/module-files
--- a/libinterp/dldfcn/module-files
+++ b/libinterp/dldfcn/module-files
@@ -1,17 +1,17 @@
 # FILE|CPPFLAGS|LDFLAGS|LIBRARIES
 __delaunayn__.cc|$(QHULL_CPPFLAGS)|$(QHULL_LDFLAGS)|$(QHULL_LIBS)
 __dsearchn__.cc
 __eigs__.cc|$(ARPACK_CPPFLAGS) $(SPARSE_XCPPFLAGS)|$(ARPACK_LDFLAGS) $(SPARSE_XLDFLAGS)|$(ARPACK_LIBS) $(SPARSE_XLIBS) $(LAPACK_LIBS) $(BLAS_LIBS)
-__fltk_uigetfile__.cc|$(GRAPHICS_CFLAGS) $(FT2_CPPFLAGS)|$(GRAPHICS_LDFLAGS) $(FT2_LDFLAGS)|$(GRAPHICS_LIBS) $(FT2_LIBS)
+__fltk_uigetfile__.cc|$(FLTK_CFLAGS) $(FT2_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS)
 __glpk__.cc|$(GLPK_CPPFLAGS)|$(GLPK_LDFLAGS)|$(GLPK_LIBS)
 __ichol__.cc
 __ilu__.cc
-__init_fltk__.cc|$(GRAPHICS_CFLAGS) $(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)|$(GRAPHICS_LDFLAGS) $(FT2_LDFLAGS)|$(GRAPHICS_LIBS) $(FT2_LIBS) $(OPENGL_LIBS)
+__init_fltk__.cc|$(FLTK_CFLAGS) $(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)|$(FLTK_LDFLAGS) $(FT2_LDFLAGS)|$(FLTK_LIBS) $(FT2_LIBS) $(OPENGL_LIBS)
 __init_gnuplot__.cc|$(FT2_CPPFLAGS) $(FONTCONFIG_CPPFLAGS)||
 __magick_read__.cc|$(MAGICK_CPPFLAGS)|$(MAGICK_LDFLAGS)|$(MAGICK_LIBS)
 __osmesa_print__.cc|$(OSMESA_CPPFLAGS) $(FT2_CPPFLAGS)|$(OSMESA_LDFLAGS)|$(OSMESA_LIBS)
 __voronoi__.cc|$(QHULL_CPPFLAGS)|$(QHULL_LDFLAGS)|$(QHULL_LIBS)
 amd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 ccolamd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 chol.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
 colamd.cc|$(SPARSE_XCPPFLAGS)|$(SPARSE_XLDFLAGS)|$(SPARSE_XLIBS)
diff --git a/libinterp/oct-conf.in.h b/libinterp/oct-conf.in.h
--- a/libinterp/oct-conf.in.h
+++ b/libinterp/oct-conf.in.h
@@ -273,16 +273,24 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_FFTW3F_LIBS
 #define OCTAVE_CONF_FFTW3F_LIBS %OCTAVE_CONF_FFTW3F_LIBS%
 #endif
 
 #ifndef OCTAVE_CONF_FLIBS
 #define OCTAVE_CONF_FLIBS %OCTAVE_CONF_FLIBS%
 #endif
 
+#ifndef OCTAVE_CONF_FLTK_CFLAGS
+#define OCTAVE_CONF_FLTK_CFLAGS %OCTAVE_CONF_FLTK_CFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_FLTK_LIBS
+#define OCTAVE_CONF_FLTK_LIBS %OCTAVE_CONF_FLTK_LIBS%
+#endif
+
 #ifndef OCTAVE_CONF_FONTCONFIG_CFLAGS
 #define OCTAVE_CONF_FONTCONFIG_CFLAGS %OCTAVE_CONF_FONTCONFIG_CFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_FPICFLAG
 #define OCTAVE_CONF_FPICFLAG %OCTAVE_CONF_FPICFLAG%
 #endif
 
@@ -305,24 +313,16 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_GLPK_LIBS
 #define OCTAVE_CONF_GLPK_LIBS %OCTAVE_CONF_GLPK_LIBS%
 #endif
 
 #ifndef OCTAVE_CONF_GNUPLOT
 #define OCTAVE_CONF_GNUPLOT %OCTAVE_CONF_GNUPLOT%
 #endif
 
-#ifndef OCTAVE_CONF_GRAPHICS_CFLAGS
-#define OCTAVE_CONF_GRAPHICS_CFLAGS %OCTAVE_CONF_GRAPHICS_CFLAGS%
-#endif
-
-#ifndef OCTAVE_CONF_GRAPHICS_LIBS
-#define OCTAVE_CONF_GRAPHICS_LIBS %OCTAVE_CONF_GRAPHICS_LIBS%
-#endif
-
 #ifndef OCTAVE_CONF_HDF5_CPPFLAGS
 #define OCTAVE_CONF_HDF5_CPPFLAGS %OCTAVE_CONF_HDF5_CPPFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_HDF5_LDFLAGS
 #define OCTAVE_CONF_HDF5_LDFLAGS %OCTAVE_CONF_HDF5_LDFLAGS%
 #endif
 
